nss-day-cohort-52/rare-client-smoky-pythons
# Refactored reactions feature to restrict a user to only one reaction per post

### Issue ticket(s): See issue tickets in corresponding [API pull request](https://github.com/nss-day-cohort-52/rare-api-smoky-pythons/pull/68)

### New file(s): N/A

### Modified File(s): *See 24 files changed in the _files changed_ tab of this PR*

### Modifications:
- Reorganized the directory/module structure
- Refactored logout and register functions to set and remove the correct token and isStaff localStorage items
- Added logic to the handlePostReaction function to restrict a user to one react per post
- Users can now remove and update existing reactions as well

### Steps to test:
1. While in the terminal, use git fetch --all in the root directory of this repo
2. Checkout out to `al-reactions-feature`
3. Confirm that your server and client are both running
4. Navigate to a post and try reacting to it...
5. You should be able to add, update, and delete you reactions
6. Try logging out of the application and register a new user...
7. When the user clicks the submit button on the RegisterUser form they should be automatically redirected to the home page. This was a bug and I think it is fixed now
